src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Digest.java

```diff
@@ -289,7 +289,7 @@ protected void engineUpdate(ByteBuffer byteBuffer) {
             }
             NIO_ACCESS.acquireSession(byteBuffer);
             try {
-                token.p11.C_DigestUpdate(session.id(), dByteBuffer.address() + ofs, null, 0, len);
+                token.p11.C_DigestUpdate(session.id(), NIO_ACCESS.getBufferAddress(byteBuffer) + ofs, null, 0, len);
```

Comment:
`long address = NIO_ACCESS.getBufferAddress(byteBuffer);` might be clearer here.